include/sparrow/typed_array.hpp

```diff
@@ -465,8 +465,25 @@ namespace sparrow
     template <class T, class Layout>
         requires is_arrow_base_type<T>
     bool operator==(const typed_array<T, Layout>& ta1, const typed_array<T, Layout>& ta2)
-    {
-        return std::equal(ta1.cbegin(), ta1.cend(), ta2.cbegin(), ta2.cend());
+    {   
+        // see https://github.com/man-group/sparrow/issues/108
+#if defined(__APPLE__) && defined(__clang__) && defined(_LIBCPP_VERSION)
+        if(ta1.size() != ta2.size())
+        { 
+            return false;
+        }
+        auto first1 = ta1.begin();
+        auto last1 = ta1.end();
+        auto first2 = ta2.begin();
+        for(; first1 != last1; ++first1, ++first2){
+            if (!(*first1 == *first2)){
+                return false;
+            }
+        }
+        return true;
+#else
+        return std::equal(ta1.begin(), ta1.end(), ta2.begin(), ta2.end());
+#endif
```

Comment:
Thanks! I think that the std::equal should work with clang 18. Could we only do the workaround for clang < 18 ?

Reply:
Correct

Reply:
Any suggestions how to detect libc++<18?

chatgtp suggests some date based versioning shenanigans for LIBCPP
`#if defined(__APPLE__) && defined(__clang__) && defined(_LIBCPP_VERSION) && (_LIBCPP_VERSION < 20220701) `

Reply:
Based on [the definition of `_LIBCPP_VERSION`](https://github.com/llvm/llvm-project/blob/main/libcxx/include/__config#L27-L30), this should work:

```c++
#if defined(__APPLE__) && defined(__clang__) && defined(_LIBCPP_VERSION) && (_LIBCPP_VERSION < 190000)
```

Reply:
My understanding is that `defined(__APPLE__) && defined(__clang__) && ` can then be removed. `libc++` can be used by other compilers than clang (like GCC and mingw, not sure about msvc) and (IMUIC) this is not specific to Apple platforms.